spec/rubocop/cop/style/trailing_body_on_method_definition_spec.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::TrailingBodyOnMethodDefinition do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) do
+    RuboCop::Config.new('Layout/IndentationWidth' => { 'Width' => 2 })
+  end
+
+  it 'registers an offense when body trails after method definition' do
+    expect_offense(<<-RUBY.strip_indent)
+      def some_method; body
+                       ^^^^ Put the method definition body on its own line.
+      end
+      def extra_large; { size: 15 };
+                       ^^^^^^^^^^^^ Put the method definition body on its own line.
+      end
+      def seven_times(stuff) 7.times { do_this(stuff) }
+                             ^^^^^^^^^^^^^^^^^^^^^^^^^^ Put the method definition body on its own line.
+      end
+    RUBY
+  end
+
+  it 'registers when body starts on def line & continues one more line' do
+    expect_offense(<<-RUBY.strip_indent)
+      def some_method; foo = {}
+                       ^^^^^^^^ Put the method definition body on its own line.
+        more_body(foo)
+      end
+    RUBY
+  end
+
+  it 'registers when body starts on def line & continues many more lines' do
+    expect_offense(<<-RUBY.strip_indent)
+      def do_stuff(thing) process(thing)
+                          ^^^^^^^^^^^^^^ Put the method definition body on its own line.
+        8.times { thing + 9 }
+        even_more(thing)
+      end
+    RUBY
+  end
+
+  it 'accepts a method with one line of body' do
+    expect_no_offenses(<<-RUBY.strip_indent)
+      def some_method
+        body
+      end
+    RUBY
+  end
+
+  it 'accepts a method with multiple lines of body' do
+    expect_no_offenses(<<-RUBY.strip_indent)
+      def stuff_method
+        stuff
+        9.times { process(stuff) }
+        more_stuff
+      end
+    RUBY
+  end
+
+  it 'does not register offense with trailing body on method end' do
+    expect_no_offenses(<<-RUBY.strip_indent)
+      def some_method
+        body
+      foo; end
```

Comment:
I'd be happy to add a sibling Cop to address this scenario. WDYT?

Reply:
👍 

Makes perfect sense to me.